src/services/codefixes/importFixes.ts

```diff
@@ -592,6 +593,15 @@ namespace ts.codefix {
                                 }
                                 break;
                             case States.NodeModules:
+                                if (fullPath.charAt(partStart + 1) === "@") {
+                                    state = States.Scope;
+                                }
+                                else {
+                                    packageRootIndex = partEnd;

```

Comment:
Another way to express this would be `goto case States.PackageContent`.  That would have the benefit of emphasizing that the two states are the same except for the initial check in `NodeModules`.